.github/workflows/build.yml

```
@@ -4,41 +4,157 @@ on:
   push:
     branches: main
   pull_request:
-    branches: "*"
+    branches: '*'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
 
 jobs:
   build:
-    runs-on: ${{ matrix.os }}
-    strategy:
-      fail-fast: false
-      matrix:
-        os: [ubuntu-latest, windows-latest, macos-latest]
-        python-version: ["3.10"]
+    runs-on: ubuntu-latest
+
     steps:
-      - name: Checkout
-        uses: actions/checkout@v2
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Base Setup
+      uses: jupyterlab/maintainer-tools/.github/actions/base-setup@v1
+
+    - name: Install dependencies
+      run: |
+        python -m pip install -U codecov
+        npm install -g codecov
+    - name: Test the extension
+      run: |
+        python -m pip install --upgrade -v -e ".[test, examples, docs]"
+        python -m pytest
+        yarn run test
```

```suggestion
        jlpm run test
```
I think with the new migration, most test use `jlpm` instead of `yarn`, this might be the cause of the failure as well.